### ISSUE TITLE
Sudden crash of spyder

### ISSUE THREAD

**Issue by VinayKillamsetty** [spyder-ide/spyder#18845]
### What steps will reproduce the problem?

first time I have experienced this crash.
It is showing Spyder has encountered an internal problem while saving the file as other file

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\ntpath.py", line 647, in realpath
    path = _getfinalpathname(path)
FileNotFoundError: [WinError 53] The network path was not found: '\\\\zeus\\Greenerwave_shared\\02_Projects\\30_Antennas\\301_Antenna_RAPID\\01_Research_and_Development\\07_Control_and_Software\\01_Antenna_Model\\02_Experimental_Fit\\01_Metasurface\\05_MultipoleModel\\OptiEvolutionist.py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\site-packages\spyder\plugins\editor\plugin.py", line 2305, in save_as
    if editorstack.save_as():
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1997, in save_as
    ao_index = self.has_filename(filename)
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1501, in has_filename
    if fixpath(filename) == fixpath(finfo.filename):
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1499, in <lambda>
    fixpath = lambda path: osp.normcase(osp.realpath(path))
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\ntpath.py", line 651, in realpath
    path = _getfinalpathname_nonstrict(path)
  File "C:\Users\VinayKILLAMSETTY\anaconda3\lib\ntpath.py", line 601, in _getfinalpathname_nonstrict
    path = _getfinalpathname(path)
FileNotFoundError: [WinError 53] The network path was not found: '\\\\zeus\\Greenerwave_shared\\02_Projects\\30_Antennas\\301_Antenna_RAPID\\01_Research_and_Development\\07_Control_and_Software\\01_Antenna_Model\\02_Experimental_Fit\\01_Metasurface\\05_MultipoleModel\\OptiEvolutionist.py'
```

## Versions

* Spyder version: 5.1.5 
* Python version: 3.9.7
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.29.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.1.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.10.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.0.4 (OK)
sphinx >=0.6.6                :  4.2.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.3 (OK)
zmq >=17                      :  22.2.1 (OK)

# Optional:
cython >=0.21                 :  0.29.24 (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.20.3 (OK)
pandas >=1.1.1                :  1.3.4 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.9 (OK)
```

**Comment by dalthviz** [comment]
Hi @VinayKillamsetty thank you for the feedback! This could be related with https://github.com/spyder-ide/spyder/issues/15714 (an operation with files that relay on a network drive). Probably, a try clause should be added with a message informing the situation instead of popping up the issue reporter.

We will try to handle this error in a future release :+1:

**Comment by dalthviz** [comment]
Checking this more I think this could have been solved on Spyde 5.3.x when fixing https://github.com/spyder-ide/spyder/issues/17685 (since there the `has_filename` was also causing the issue when trying to check a file over a network drive/shared folder). @VinayKillamsetty could you update to Spyder 5.3.2 and check again? For that you will need to create a new env using the conda-forge channel by running something like this from an Anaconda prompt:

   conda create -n spyder-cf -c conda-forge spyder
   conda activate spyder-cf
   spyder

You can also launch Spyder after creating the env by using the application shortcut (to use it you can go to the start menu and search `Spyder (spyder-cf)`)

Let us know if that helps!

**Comment by dalthviz** [comment]
Closing due to lack of response
